### ISSUE TITLE
How to test a whole dataset?

### ISSUE THREAD

**Issue by 978326187** [uzh-rpg/rpg_asynet#9]
i notice that the **sparse_VGG_test.py**  just test one image in NCaltech101 , how do i test the accuracy in the whole dataset?

**Comment by 978326187** [comment]
if we use the histogram generate the data flow, we can't use the time-stamp in event.

**Comment by MessikommerNico** [comment]
The file sparse_VGG_test.py is a unittest to ensure that the output of SSC (facebook implementation) is identical to our asynchronous asynet. It does not test the prediction accuracy.
In order to avoid having too many files, we excluded many additional files including the file for the test evaluation.
To obtain the test accuracy, a quick and dirty hack would be to change the validation dataset path in the datalooder to the test dataset path and start the training script.

**Comment by 978326187** [comment]
i wanna know how many epoch i need to achieve the result of the paper,  i have trained 100 epoch using **fb_sparse_vgg** model, and the accuracy is just 0.6 in N-Caltech101.

**Comment by MessikommerNico** [comment]
The accuracy after 100 epochs should be around 0.65. The test accuracy reported in the paper was obtained with a model trained for 1500 epochs.

**Comment by 978326187** [comment]
ok, thanks for your reply~

**Comment by MessikommerNico** [comment]
It depends on the specific hardware implementation. With a smart memory access design, the asynchronous sparse convolutions can be much faster as the computational complexity is much lower compared to normal convolutions.

**Comment by 978326187** [comment]
It would be fantastic if the specific hardware is created.

**Comment by 978326187** [comment]
I just remembered what I wanted to ask is How to test a whole dataset using **asynet** . we can only using SSC in training  script

**Comment by MessikommerNico** [comment]
The output of SSC and asynet is identical. Thus, the accuracy achieved by SSC is the same as achieved by asynet
